app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java

```diff
@@ -572,11 +572,27 @@ public Boolean execute(SFTPClient client) {
                 .getFolder();
         break;
       case GDRIVE:
-        isDirectory =
-            dataUtils
-                .getAccount(OpenMode.GDRIVE)
-                .getMetadata(CloudUtil.stripPath(OpenMode.GDRIVE, path))
-                .getFolder();
+        final boolean[] isDir = new boolean[1];
+        Thread googleThread = new Thread(new Runnable() {
+          @Override
+          public void run() {
+            isDir[0] = dataUtils
+                            .getAccount(OpenMode.GDRIVE)
+                            .getMetadata(CloudUtil.stripPath(OpenMode.GDRIVE, path))
+                            .getFolder();
+          }
+        });
+        googleThread.start();
+        try{
+          googleThread.join();
```

Comment:
No joining threads in main thread

Reply:
Any suggestions? I can't find a way to change it.

Reply:
The new code has update.